node/cn/backend.go

```diff
@@ -453,48 +443,6 @@ func New(ctx *node.ServiceContext, config *Config) (*CN, error) {
 	return cn, nil
 }
 
-func (s *CN) createSnapshot() {
-	var (
-		currBlock = s.blockchain.CurrentBlock()
-		headers   []*types.Header
-		err       error
-	)
-	// Temporarily supply blockchain for `Finalize`, staking module for `snapshot` and reward module for `processor`.
-	s.blockchain.Engine().(consensus.Istanbul).SetChain(s.blockchain)
-	mStaking := staking_impl.NewStakingModule()
-	mReward := reward_impl.NewRewardModule()
-	mStaking.Init(&staking_impl.InitOpts{
-		ChainKv:     s.chainDB.GetMiscDB(),
-		ChainConfig: s.chainConfig,
-		Chain:       s.blockchain,
-	})
-	mReward.Init(&reward_impl.InitOpts{
-		ChainConfig:   s.chainConfig,
-		Chain:         s.blockchain,
-		GovModule:     s.govModule,
-		StakingModule: mStaking,
-	})
-	s.blockchain.Engine().(consensus.Istanbul).RegisterStakingModule(mStaking)
-	s.blockchain.Engine().(consensus.Istanbul).RegisterConsensusModule(mReward)
-	if headers, err = s.Engine().GetKaiaHeadersForSnapshotApply(s.blockchain, currBlock.NumberU64(), currBlock.Hash(), nil); err != nil {
-		logger.Error("Failed to get headers to apply", "err", err)
-	} else {
-		preloadRef, err := reward.PreloadStakingInfo(s.blockchain, headers, mStaking)
```

Comment:
We can delete reward/staking_manager.go:PreloadStakingInfo and related types as well.

Reply:
deleted https://github.com/kaiachain/kaia/pull/190/commits/3e10c2d4ed664226bcca62c31d0fcdf76137448a